### DISCUSSION TITLE
Testing routes

### DISCUSSION THREAD

**Discussion by anthonyjmartinez** [routerify/routerify#70] [Q&A]
Is there a standard way to write unit tests for a specific route? I'm working on a library that includes several handlers, and would like to be able to test that they do what I expect within `mod tests` but I'm a little lost on how I might accomplish that. I see that `routerify` itself defines a `support` module inside its integration tests - should I try to re-implement that myself in my own project?

Thanks!

**Comment by seanpianka** [comment]
You'll need to setup the test infra code to write these tests, like building the request + body and calling the request service. You can see an example of using the `RequestServiceBuilder` to instantiate a `RequestService` and call it using a `http::Request<hyper::Body>` [here](https://github.com/routerify/routerify/blob/master/examples/aws_lambda.rs).

This is an alternative to the setup method you see in the `support` module. Either way works, just choose one more suitable to you.
